### DISCUSSION TITLE
b.plot work but b.plot_distribution_collection show nothing

### DISCUSSION THREAD

**Discussion by famc2** [phoebe-project/phoebe2#641] [Questions]
fresh installed phoebe 2.4.5
b.plot work, but
I followed http://phoebe-project.org/docs/2.4/tutorials/distribution_propagation
nothing showed

I have test matplotlib 3.5.2 3.5.0 in python 3.9 3.8

Thank you!

**Comment by kecnry** [comment]
Are you running in a jupyter notebook or interactive python session?  If you're in a notebook, `%matplotlib inline` at the top of the notebook might do the trick (although this probably isn't the cause if `b.plot` is showing the plot).

Does `b.plot_distribution_collection(...)` return a matplotlib figure instance?  Can you get that to show or save to an image successfully?

If none of those work, can you try downgrading matplotlib below 3.5.0 and letting us know if that helps?  There are some other reported matplotlib compatibility problems (#636), but most of those so far are only with 3.5.2+.

Thanks!
Kyle

**Reply by famc2** [reply]
in this tutorial: 
http://phoebe-project.org/docs/2.4/tutorials/emcee
after this line
_ = b.plot(solution='emcee_sol', style='lnprobability', show=True)
the previous b.plot_distribution_collection showed.

I use python command not jupyter.
b.plot_distribution_collection(...) return nothing.

I have downgrade matplotlib to 3.5.0, but still not work.

thank you for quick reply!
famc2

**Reply by kecnry** [reply]
hmm, that is suggests that it might not directly be a plotting dependency issue if previous calls did work.  Does `b.get_distribution_collection()` (with the same arguments except for `show`) return something?  Does calling `.plot(...)` on that returned object work?

**Reply by famc2** [reply]
`print(b.get_distribution_collection(distribution='ball_around_guess')) ` return:
(<phoebe.dependencies.distl.distl.DistributionCollection object at 0x7fc354c78fd0>, ['sma@binary@orbit@component', 'incl@binary@orbit@component', 't0_supconj@binary@orbit@component', 'requiv@primary@star@component', 'pblum@primary@lc01@lc@dataset', 'sigmas_lnf@lc01@lc@dataset'])


dc, twigs = b.get_distribution_collection(distribution='ball_around_guess') 
dc.dists is:
[<distl.gaussian loc=7.01 scale=0.1 unit=solRad label=sma@binary label_latex=$a_\mathrm{ binary }$>, <distl.gaussian loc=80.1 scale=5.0 unit=deg label=incl@binary label_latex=$i_\mathrm{ binary }$>, <distl.gaussian loc=0.099 scale=0.001 unit=d label=t0_supconj@binary label_latex=$t_{0, \mathrm{ supconj }, \mathrm{ binary } }$>, <distl.gaussian loc=1.95 scale=0.4 unit=solRad label=requiv@primary label_latex=$R_{ \mathrm{ equiv }, \mathrm{ primary } }$>, <distl.gaussian loc=12.566370614359172 scale=0.2 unit=W label=pblum@primary@lc01 label_latex=$L_\mathrm{ pb, lc01 }$>, <distl.uniform low=-1000000000.0 high=-10000.0 unit= label=sigmas_lnf@lc01 label_latex=$\sigma_\mathrm{ lnf, lc01 }$>]

but 
dc.plot()
show nothing

**Reply by kecnry** [reply]
ok, then I really think it is some sort of plotting issue itself, but it is really difficult to debug.  Can you try either skipping the previous plotting commands as you run the notebook and then see if this one shows (to make the distinction between something in this particular call vs any Nth plotting call failing to show)?  Can you try running in a python script instead of a notebook and see if the plot shows in a popup window?

**Comment by famc2** [comment]
Hi, I've been running python script, not jupyter.
I add plt.close() before plot() and plt.show() after plot(), then everything OK.
reference:
https://stackoverflow.com/questions/51568030/plt-figure-figure-show-does-nothing-when-not-executing-interactively

**Reply by kecnry** [reply]
very interesting, we'll keep that in mind if anyone else runs into something similar.  Thanks for the update!
